uberfire-structure/uberfire-structure-backend/src/main/java/org/guvnor/structure/backend/pom/PomEditorDefault.java

```diff
@@ -53,58 +62,85 @@ public boolean addDependency(DynamicPomDependency dep,
             if (!keys.containsKey(keyDep)) {
+                Dependency pomDep = getMavenPomDep(dep,
+                                                   internalArtifact,
+                                                   getKieVersionFromPom(dep,
+                                                                        depsFromPom));
                 model.getDependencies().add(pomDep);
+                result = true;
             } else {
-                //override dep version with the version contained in the json
-                String versionKey = keys.get(keyDep);
-                List<Dependency> modelDeps = model.getDependencies();
-                for (Dependency modelDep : modelDeps) {
+                for (Dependency modelDep : depsFromPom) {
                     if (modelDep.getGroupId().equals(dep.getGroupID()) && modelDep.getArtifactId().equals(dep.getArtifactID())) {
-                        modelDep.setVersion(versionKey);
+                        if (internalArtifact) {
+                            modelDep.setVersion(getKieVersionFromPom(dep,
+                                                                     depsFromPom));
+                        } else {
+                            modelDep.setVersion(keys.get(keyDep));
+                        }
                     }
                 }
-                return false;
             }
-            writePOMModelOnFS(filePath,
-                              model);
+            if (result) {
+                writePOMModelOnFS(filePath,
+                                  model);
+            }
         } catch (Exception ex) {
             logger.error(ex.getMessage(),
                          ex);
-            return false;
+            result = false;
         }
-        return true;
+        return result;
     }
 
-    public boolean addDependencies(List<DynamicPomDependency> deps,
+    public boolean addDependencies(Set<DependencyType> dependencyTypes,
                                    Path pomPath) {
+        List<DynamicPomDependency> deps = new ArrayList<>();
+        for (DependencyType dep : dependencyTypes) {
+            deps.addAll(mapper.getDependencies(EnumSet.of(dep)));
+        }
+
         if (deps.isEmpty()) {
             return false;
         }
         boolean result = false;
         try {
             org.uberfire.java.nio.file.Path filePath = Paths.get(pomPath.toURI());
             Model model = getPOMModel(filePath);
-            Map<String, String> keys = getKeysFromDeps(model.getDependencies());
+            List<Dependency> depsFromPom = model.getDependencies();
+            Map<String, String> keys = getKeysFromDependencies(model.getDependencies());
 
             for (DynamicPomDependency dep : deps) {
-                String keyDep = getKeyFromDep(dep);
-                if (!keys.containsKey(keyDep)) {
-                    Dependency pomDep = getMavenDependency(dep);
+                boolean internalArtifact = isInternalArtifact(dep);
```

Comment:
also this code looks like a duplication of addDependency right? Could you please do a refactoring?